src/commands.cpp

```diff
@@ -236,63 +240,53 @@ CON_COMMAND_CHAT(ztele, "teleport to spawn")
 	while (nullptr != (spawn = (SpawnPoint*)UTIL_FindEntityByClassname(spawn, "info_player_")))
 	{
 		if (spawn->m_bEnabled())
-		{
 			spawns.AddToTail(spawn);
-		}
 	}
 
 	//Pick and get position of random spawnpoint
 	int randomindex = rand() % spawns.Count()+1;
 	Vector spawnpos = spawns[randomindex]->GetAbsOrigin();
 
 	//Here's where the mess starts
-	CBasePlayerPawn* pPawn = player->m_hPawn();
+	CBasePlayerPawn* pPawn = player->GetPawn();
+
 	if (!pPawn)
-	{
 		return;
-	}
-	if (pPawn->m_iHealth() <= 0)
+
+	if (!pPawn->IsAlive())
 	{
 		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX"You cannot teleport when dead!");
 		return;
 	}
+
 	//Get initial player position so we can do distance check
 	Vector initialpos = pPawn->GetAbsOrigin();
 
 	ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX"Teleporting to spawn in 5 seconds.");
 
-	//Convert into handle so we can safely pass it into the Timer
-	auto handle = player->GetHandle();
-	new CTimer(5.0f, false, false, [spawnpos, handle, initialpos]()
-		{
-			//Convert handle into controller so we can use it again, and check it isn't invalid
-			CCSPlayerController* controller = (CCSPlayerController*)Z_CBaseEntity::EntityFromHandle(handle);
-
-			if (!controller || controller->m_iConnected() != PlayerConnectedState::PlayerConnected)
-				return;
+	CHandle<CBasePlayerPawn> handle = pPawn->GetHandle();
 
-			//Get pawn (again) so we can do shit
-			CBasePlayerPawn* pPawn2 = controller->m_hPawn();
+	new CTimer(5.0f, false, false, [spawnpos, handle, initialpos]()
+	{
+		if (!handle.Get())
+			return;
 
-			//Get player origin after 5secs
-			Vector endpos = pPawn2->GetAbsOrigin();
+		CBasePlayerPawn *pPawn = handle.Get();
```

Comment:
Why are we using Get()? I believe this will not properly check the validity of a handle, hence the existance of EntityFromHandle

Reply:
Good catch

Reply:
Actually upon closer inspection, Get() ultimately does compare the handle with the retrieved entity's handle, which would make `EntityFromHandle` redundant.

Reply:
ah I see, great. Yeah we can get rid of EntityFromHandle then